source/module_hsolver/test/diago_lapack_test.cpp

```diff
@@ -0,0 +1,368 @@
+// Author: Zhang Xiaoyang
+// A modified version of diago_lcao_test.cpp
+// Remove some useless functions and dependencies. Serialized the full code
+// and refactored some function.
+
```

Comment:
Hi @Critsium-xy, in diago_lcao_test.cpp, I use the results of lapack as the reference to test elpa and scalapack. But, here you also use the lapack as the reference of DiagoLapack in abacus, which may only test whether the call to the lapack interface is correct. I suspect you use the results of other diagonalization method (such as the results by NumPy or Matlab) as the reference, or even a matrix solved by yourself.

Reply:
ok, I will modify it later

Reply:
I added the solution file under tests folder. Then I will use the solutions read from file to compare with the caculated result.